packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -871,21 +871,23 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
       const isRangeSelected = selectionStart !== selectionEnd;
       const inputValue = element.value;
 
-      // We shouldn't lose focus in the following cases:
+      // If the element is enabled, we should lose focus except when:
       // 1. There is range selected.
       // 2. When selection start is larger than 0 and it is backward.
       // 3. when selection start is not the end of length and it is forward.
       // 4. We press any of the arrow keys when our handleTabKey isn't none or undefined (only losing focus if we hit tab)
       // and if shouldInputLoseFocusOnArrowKey is defined, if scenario prefers to not loose the focus which is determined by calling the
       // callback shouldInputLoseFocusOnArrowKey
-      if (
-        isRangeSelected ||
-        (selectionStart! > 0 && !isForward) ||
-        (selectionStart !== inputValue.length && isForward) ||
-        (!!this.props.handleTabKey &&
-          !(this.props.shouldInputLoseFocusOnArrowKey && this.props.shouldInputLoseFocusOnArrowKey(element)))
-      ) {
-        return false;
+      if (element.readOnly === false) {
```

Comment:
We definitely shouldn't do this, it's making an assumption that could (and does) break scenarios where inputs are expected to get focus

Reply:
If you want this functionality, why don't you use `shouldInputLoseFocusOnArrowKey`?

Reply:
[shouldInputLoseFocusOnArrowKey](https://github.com/OfficeDev/office-ui-fabric-react/blob/cfa7a19ec1d6f98c13f822a828e0751f64507dbd/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L885-L886) requires a non-zero flag be set for HandleTabKey.

The issue is that these [two lines](https://github.com/OfficeDev/office-ui-fabric-react/blob/cfa7a19ec1d6f98c13f822a828e0751f64507dbd/packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx#L883-L884) don't really work for disabled inputs, as selectionStart will always be 0. I could move the readOnly check to short-circuit those two conditions.

So instead of it wrapping the whole whole if-block in the readOnly check, it would look like this:

      if (
        isRangeSelected ||
        (element.readOnly !== false && (selectionStart! > 0 && !isForward)) ||
        (element.readOnly !== false && (selectionStart !== inputValue.length && isForward)) ||
        (!!this.props.handleTabKey &&
          !(this.props.shouldInputLoseFocusOnArrowKey && this.props.shouldInputLoseFocusOnArrowKey(element)))
      ) {
        return false;
      }

Would that work for the scenarios you're referring to, @jspurlin?

Reply:
I believe this bug goes away if the `input` element gets a disabled attribute. So there is some "is element focusable" logic somewhere that is looking for that. It should also look for `aria-disabled`, which IS present. I think the right fix is to add the `aria-disabled` check to that logic.